### ISSUE TITLE
Connection failures of remote reporters

### ISSUE THREAD

**Issue by SkyterX** [Recognos/Metrics.NET#53]
Currently, when any of remote reporters (ElasticSearch, Graphite, InfluxDB, etc.) loses connection to the host and sending metrics starts throwing exceptions, its task is eventually killed by ScheduledReporter (depending on allowed number of errors). 
After that, the only way to restore reporting is to restart whole application.
This is a pretty typical scenario, for example, when the remote host is shut down for maintenance.
Moreover, the host application has no way to verify, if the remote reporter is working, to be able to restart the task automatically.

**Comment by maxcherednik** [comment]
Wasn't there a -1 setting to retry eternally?

**Comment by SkyterX** [comment]
Yes, there is this setting, but there are still some issues with how it's implementation:
1. The default value of this setting is 0 (break on first error), while the value -1 (retry forever) seems much more reasonable in a production environment. Network errors happen, and you don't expect your metrics to break because of that. Adding this setting for graphite reporting was a breaking change. The value 0 can be reasonable in local reporters when you don't expect the problem to cease over time, which brings us to the second issue.
2. The setting is global. There is no way to configure error count thresholds for different reporters. Moreover, the only way to configure it is via application settings, and there is no simple way to share settings among projects.
3. The host application has no proper way to handle this kind of errors. You can set up some custom error handler, to handle exceptions inside Metric.NET, but an exception itself doesn't provide enough context of where it comes from. 
4. There is no proper way to restart failed reporters. Even if you've managed to fix the error, to resume reporting of metrics you'll have to restart your application.
